src/App.js

```diff
@@ -17,34 +17,36 @@ import Welcome from './Welcome';
 function App() {
   const [token, setToken] = useState(localStorage.getItem('userToken'));
   const [list, setList] = useState([]);
-  const [loading, setLoading] = useState(true);
+  const [loading, setLoading] = useState(false);
   const [error, setError] = useState(null);
 
   useEffect(() => {
-    let unsubscribe;
-    try {
-      unsubscribe = db.collection(token).onSnapshot((snapshot) => {
-        const newList = [];
-        snapshot.forEach((doc) => {
-          // This seems to work, while snapshot.map doesn't
-          newList.push(
-            JSON.stringify(doc.data()['formData']['itemName']).replace(
-              /['"]+/g,
-              '',
-            ),
-          );
-        });
-        setList(newList);
-        setError(null);
+    const unsubscribe = () => {
+      try {
+        token
```

Comment:
@dariatsvetkova Here I'm only calling the db if I have a token. Otherwise I set an error telling the user they need a shopping list.

I also updated to `setLoading(true)` only when we are actually calling the db.

Reply:
Deceptively simple!